app.R

```diff
@@ -0,0 +1,270 @@
+library(shiny)
+library(ggplot2)
+library(leaflet)
+library(tidyverse)
+library(bslib)
+library(plotly)
+library(thematic)
+library(countrycode)
+library(DT)
+library(geojsonio)
+library(htmltools)
+
+options(shiny.autoreload = TRUE)
+
+# Read the data and clean it
+data <- read_csv("data/hotel_bookings.csv",
+                 show_col_types = FALSE)
+
+data <- data[!is.na(data$country), ]
+
+data <- data |> mutate(
+  country_full_name = case_when(
+    !country %in% c("CN", "TMP") ~ countrycode(country, "iso3c", "country.name"),
+    country == "CN" ~ "China",
+    country == "TMP" ~ "East Timor"
+  )
+)
+data <- data[!is.na(data$country_full_name), ]
+data$country_full_name[data$country_full_name == 'United States'] <- "United States of America"
+data$country_full_name[data$country_full_name == 'Czechia'] <- "Czech Republic"
+data$country_full_name[data$country_full_name == 'Bosnia & Herzegovina'] <- "Bosnia and Herzegovina"
+data$country_full_name[data$country_full_name == 'Serbia'] <- "Republic of Serbia"
+data$country_full_name[data$country_full_name == 'North Macedonia'] <- "Macedonia"
+
+# Read the map polygon data
+countries_json <- geojson_read("data/countries.geojson", what = "sp")
+country_list = pull(countries_json@data[1])
+full_country_df = data_frame(country_full_name = country_list)
+
+ui <- fluidPage(
+  theme = bs_theme(bootswatch = 'flatly'),
+  # Page title 
+  titlePanel(h2('🏨 🧐 Hotel Industry Competitive Landscape Dashboard',
+                align = "center")),
+  br(),
+  
+  # Controller row
+  fluidRow(
+    dateRangeInput(inputId = 'daterange',
+                   label = '🗓️ Date range:',
+                   start  = min(data$reservation_status_date),
+                   end = max(data$reservation_status_date),
+                   format = "mm/dd/yyyy"
+                   ),
+    selectInput(inputId = 'countires',
```

Comment:
Is this meant to be "countires" as the inputID?

Reply:
yeah, sorry for the typo. I'll fix it now.